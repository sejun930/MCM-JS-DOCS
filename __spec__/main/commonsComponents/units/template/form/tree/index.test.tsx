import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";

import _TreeForm from "src/main/commonsComponents/units/template/form/tree";

export default describe("Tree Template Page", () => {
  // 스냅샷
  test("Snapshot", () => {
    const { container } = render(
      <RecoilRoot>
        <_TreeForm />
      </RecoilRoot>
    );
    expect(container).toMatchSnapshot();
  });

  // subTitle 태그의 존재여부 검증
  test("Check h2 tag and p tag", () => {
    const { container } = render(
      <RecoilRoot>
        <_TreeForm />
      </RecoilRoot>
    );

    const wrapperEle = container.querySelector(".tree-subTitle ");
    const h2Ele = wrapperEle?.querySelector(".mcm-title-unit");
    expect(h2Ele).toBeInTheDocument();

    if (h2Ele) {
      expect(h2Ele.textContent).toEqual("📂 Module Tree");
    }

    const remarksEle = container.querySelector(".subTitle-remarks");
    expect(remarksEle).toBeInTheDocument();

    if (remarksEle) {
      expect(remarksEle.textContent).toEqual(
        "모듈의 구조 안의 각각의 태그들이 어떤 역할을 하고 있는지 태그를 클릭해보세요."
      );
    }
  });

  // list, detail 페이지가 있는지를 검증
  test("Check list and detail wrapper page", () => {
    const { container } = render(
      <RecoilRoot>
        <_TreeForm />
      </RecoilRoot>
    );

    // list 페이지가 있는지 검증
    const listWrapperEle = container.querySelector(".tree-list-wrapper");
    expect(listWrapperEle).toBeInTheDocument();

    // detail 페이지가 있는지 검증
    const detailWrapperEle = container.querySelector(".tree-detail-wrapper");
    expect(detailWrapperEle).toBeInTheDocument();
  });
});
