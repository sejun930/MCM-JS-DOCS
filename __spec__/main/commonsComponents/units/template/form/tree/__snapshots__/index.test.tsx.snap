// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree Template Page Snapshot 1`] = `
<div>
  <div
    class="css-1iyoj2o"
  >
    <div
      class="subTitle-Template tree-subTitle css-cw3rkw"
    >
      <div
        class="subTitle-title-wrapper css-1knbux5"
      >
        <h2
          class="mcm-title-unit subTitle"
        >
          📂 Module Tree
        </h2>
      </div>
      <div
        class="subTitle-line css-18klhw4"
      />
      <p
        class="mcm-p-unit subTitle-remarks css-z401r8"
      >
        모듈의 구조 안의 각각의 태그들이 어떤 역할을 하고 있는지 태그를 클릭해보세요.
      </p>
    </div>
    <div
      class="css-1757l5l"
    >
      <div
        class="tree-list-wrapper css-1hxre6i"
      >
        <div
          class="copy-wrapper tree-list css-di3nu0"
        >
          <div
            class="copy-text css-gjwso2"
          >
            <pre
              class="copy-code-list css-1gnhfxs"
            >
              <code
                class="copy-code"
              />
            </pre>
          </div>
        </div>
      </div>
      <div
        class="tree-detail-wrapper css-zmto2j"
      >
        <div
          class="tree-detail-empty-select css-12hwwz"
        >
          <p
            class="mcm-p-unit css-z401r8"
          >
            태그를 클릭해보세요.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
